client/lib/domains/dns/index.js

```diff
@@ -126,7 +130,57 @@ function getFieldWithDot( field ) {
 	return ( typeof field === 'string' && field.match( /^([a-z0-9-_]+\.)+\.?[a-z]+$/i ) ) ? field + '.' : field;
 }
 
-module.exports = {
-	validateAllFields,
-	getNormalizedData
+function isWpcomRecord( record ) {
+	return startsWith( 'wpcom:', record.id );
```

Comment:
I found a bug/regression - when you add a new custom A record, the old wpcom A record is not removed from the view (on the backend everything is saved A-OK, according to my tests). Since I felt bad about delaying this PR again with my other comments (and I was feeling curious), I dug deeper. Turns out, one of my suggestions caused this regression - well, kind of. `startsWith` is the culprit. Of course, it works as it should but... the order of the arguments is [the other way around](https://lodash.com/docs#startsWith) XD So this always return `false`, bla bla bla, you get the picture :)